go/store/nbs/journal_writer.go

```diff
@@ -187,36 +188,67 @@ func (wr *journalWriter) bootstrapJournal(ctx context.Context) (last hash.Hash,
 		if info, err = wr.index.Stat(); err != nil {
 			return hash.Hash{}, err
 		}
+
+		// initialize range index with enough capacity to
+		// avoid rehashing during bootstrapping
+		cnt := estimateRangeCount(info)
+		wr.ranges.cached = swiss.NewMap[addr16, Range](cnt)
+
+		eg, ectx := errgroup.WithContext(ctx)
+		ch := make(chan []lookup, 4)
+
 		// process the indexed portion of the journal
-		err = processIndexRecords(ctx, wr.index, info.Size(), func(o int64, r indexRec) (err error) {
-			switch r.kind {
-			case tableIndexRecKind:
-				// |r.end| is expected to point to a root hash record in |wr.journal|
-				// containing a hash equal to |r.lastRoot|, validate this here
-				var h hash.Hash
-				if h, err = peekRootHashAt(wr.journal, int64(r.end)); err != nil {
```

Comment:
`peekRootHash` uses a fixed size buffer of 1024 bytes and does not slice the buffer before passing it to `validateIndexRecord`, so the checksum step always failed.
https://github.com/dolthub/dolt/blob/29e0f200122d801f78f4c29a53d80ffb28a82142/go/store/nbs/journal_record.go#L253